app/Http/Controllers/PollerGroupController.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Device;
+use App\Models\Poller;
+use App\Models\PollerGroups;
+use App\Models\PollerCluster;
+use Illuminate\Validation\Rule;
+use Illuminate\Http\Request;
+use LibreNMS\Alerting\QueryBuilderFilter;
+use LibreNMS\Alerting\QueryBuilderFluentParser;
+use LibreNMS\Config;
+use Toastr;
+
+class PollerGroupController extends Controller
+{
+    public $rrdstep;
+    public $isDistributedPoller;
+    public $defaultPollerId;
+
+    public $defaultPollerTab = 'poller';
+
+    public $defaultGroup = ['id' => 0,
+                             'group_name' => 'General',
+                             'descr' => ''];
+    public $defaultPollerMarker = '(default Poller)';
+
+    public function __construct()
+    {
+        $this->authorizeResource(PollerGroup::class, 'poller_groups');
+
+        $this->rrdstep = \LibreNMS\Config::get('rrd.step');
+
+        $this->isDistributedPoller = \LibreNMS\Config::get('distributed_poller');
+        $this->defaultPollerId = \LibreNMS\Config::get('distributed_poller_group');
+    }
+
+    public function index(Request $request)
+    {
+        $current_tab = $request['tab'] ?: $this->defaultPollerTab;
+
+        switch ($current_tab) {
+            case 'poller':
+                $data = $this->pollerTab();
+                break;
+            case 'groups':
+                $data = $this->groupsTab();
+                break;
+            case 'performance':
+                $data = [];
+                break;
+            case 'log':
+                $data = $this->logTab($request);
+                break;
+            default:
+                $current_tab = $this->defaultPollerTab;
+                $data = $this->pollerTab();
+                break;
+        }
+
+        return view('poller-group.'.$current_tab, array_merge($data, ['current_tab' => $current_tab]));
+    }
+
+    public function logTab($request)
+    {
+        return ['filter' => $request['filter'] ?: 'active'];
```

Comment:
You can also use $request like this `$request->input('filter', 'active')`

Reply:
changed to it